### PR TITLE
perf(agg/foldable): implement `BitmapOnesIter` and use it in apply_batch_concrete()

### PR DESCRIPTION
**Pull request by Honeta**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

as title

## Checklist For Contributors

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Checklist For Reviewers

- [ ] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.

## Documentation

- [x] My PR **DOES NOT** contain user-facing changes.

benchmark (unit: ns/iter)

| case                                     | baseline | current |    diff   |
|:----------------------------------------:|:--------:|:-------:|:---------:|
| bench_foldable_count_agg_vis_rate_0_05   |  189,442 |  87,946 |  -53.56%  |
| bench_foldable_count_agg_vis_rate_0_25   |  209,099 | 110,386 |  -47.19%  |
| bench_foldable_count_agg_vis_rate_0_5    |  236,969 | 148,230 |  -37.47%  |
| bench_foldable_count_agg_vis_rate_0_75   |  271,424 | 189,227 |  -30.27%  |
| bench_foldable_count_agg_without_vis     |  129,485 | 129,469 |  -0.012%  |
| bench_foldable_max_agg_vis_rate_0_05     |  219,781 |  82,210 |  -62.60%  |
| bench_foldable_max_agg_vis_rate_0_25     |  232,116 | 128,172 |  -44.77%  |
| bench_foldable_max_agg_vis_rate_0_5      |  267,885 | 197,163 |  -26.38%  |
| bench_foldable_max_agg_vis_rate_0_75     |  288,743 | 272,015 |  -5.79%   |
| bench_foldable_max_agg_without_vis       |  257,916 | 229,613 |  -10.97%  |
| bench_foldable_min_agg_vis_rate_0_05     |  220,225 |  89,658 |  -59.28%  |
| bench_foldable_min_agg_vis_rate_0_25     |  235,350 | 138,322 |  -41.26%  |
| bench_foldable_min_agg_vis_rate_0_5      |  261,232 | 212,706 |  -18.58%  |
| bench_foldable_min_agg_vis_rate_0_75     |  289,783 | 281,721 |  -2.78%   |
| bench_foldable_min_agg_without_vis       |  229,307 | 229,185 |  -0.053%  |
| bench_foldable_sum_agg_vis_rate_0_05     |  235,178 |  85,474 |  -63.65%  |
| bench_foldable_sum_agg_vis_rate_0_25     |  301,981 | 177,898 |  -41.03%  |
| bench_foldable_sum_agg_vis_rate_0_5      |  405,590 | 304,585 |  -24.90%  |
| bench_foldable_sum_agg_vis_rate_0_75     |  538,044 | 445,109 |  -17.29%  |
| bench_foldable_sum_agg_without_vis       |  609,232 | 620,220 |  +1.80%   |

### CODE REVIEW

**Comment by BugenZhao** [comment]
I guess we should instead refactor the implementation of `iter`? 🤔 It's not elegant to expand the iteration everywhere.

**Comment by Honeta** [comment]
> I guess we should instead refactor the implementation of `iter`? 🤔 It's not elegant to expand the iteration everywhere.

I think the problem here is that using enumerate() after using iter_ones() can't get the original index, so it can't meet the needs of this part of the code. There may be other solutions to this problem.

**Review by st1page — COMMENTED** [review]
Let's add a new kind of Bitmap's iterator which is implemented by `next_set_bit` and then use it to implement `Bitmap::iter_ones()`. You can refer the `DataChunkRefIter`. it might be like
```rust
pub struct Ones<'a> {
    bitmap: &'a Bitmap,
    /// `None` means finished
    idx: Option<usize>,
}

impl<'a> Iterator for Ones<'a> {
    type Item = usize;

    fn next(&mut self) -> Option<usize> {
     }
```

File: src/common/src/buffer/bitmap.rs [hunk 1]
```diff
@@ -629,6 +629,24 @@ impl<'a> iter::Iterator for BitmapIter<'a> {
 impl ExactSizeIterator for BitmapIter<'_> {}
 unsafe impl TrustedLen for BitmapIter<'_> {}
 
+pub struct BitmapOnesIter<'a> {
+    bitmap: &'a Bitmap,
+    /// `None` means finished
```
Comment (st1page) [inline src/common/src/buffer/bitmap.rs]:
```suggestion
```

**Review by st1page — APPROVED** [review]

File: src/common/src/buffer/bitmap.rs (hunk 1 above)
Comment (Honeta) [reply src/common/src/buffer/bitmap.rs]: already removed, display bug

**Review by st1page — COMMENTED** [review]
Please add some unit tests because and we need to optimize the iter later. Thanks!

**Review by st1page — APPROVED** [review]

**Pull request merged**